projects/fsastorefrontlib/src/cms-components/my-account/claim/change-claim/change-claim-navigation.component.ts

```diff
@@ -0,0 +1,71 @@
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { RoutingService } from '@spartacus/core';
+import { FileService, FormDataService } from '@spartacus/dynamicforms';
+import { ClaimService } from '../../../../core/my-account/facade/claim.service';
+import { combineLatest, Subscription } from 'rxjs';
+import { map, tap } from 'rxjs/operators';
+
+@Component({
+  selector: 'cx-fs-change-claim-navigation',
+  templateUrl: './change-claim-navigation.html',
+})
+export class ChangeClaimNavigationComponent implements OnInit, OnDestroy {
+  constructor(
+    protected routingService: RoutingService,
+    protected claimService: ClaimService,
+    protected fileService: FileService,
+    protected formDataService: FormDataService
+  ) {}
+
+  claim$;
+  subscription = new Subscription();
+
+  ngOnInit(): void {
+    this.subscription.add(
+      this.routingService
+        .getRouterState()
+        .pipe(
+          tap(routingData => {
+            const claimId = routingData.state.params.claimId;
+            if (claimId) {
```

Comment:
You also need this check here `&& !routingData.nextState`, because as I can see you have `back` button/link which will trigger `loadClaimById ` when you click on back and we should `loadClaimById` only once when we come to the page and not also when we leave the page.

Reply:
Or just use filter operator before tap `filter(routingData => !routingData.nextState)`